usr/sbin/p11sak/p11sak.c

```diff
@@ -2678,7 +2678,7 @@ static CK_RV list_ckey(CK_SESSION_HANDLE session, CK_SLOT_ID slot,
             fprintf(stderr, "Retrieval of label failed (error code 0x%lX: %s)\n", rc,
                     p11_get_ckr(rc));
         } else if (long_print) {
-            printf("Label: %s\t\t", label);
+            printf("Label: \"%s\"\t\t", label);
```

Comment:
Looks good now, thanks!

Unfortunately it still does not solve one case, where the objects label is padded with one zero byte, e.g. `Wrap_Key\0` and ulValueLen includes the zero-byte (e.g.  ulValueLen = 9). One of our testcases creates such key objects.... ;-) 

When printing that label, that zero-byte is treated as string end, and it is thus not printed. So the quoting does not help here. We could certainly add some magic into `tok_key_get_label_attr()` to format the returned label string in a way that it escapes zero bytes as `\0` if they are included in the ulValueLen. With that it would print as `"Wrap_Key\0"`.

However, even with that, one would not be able to specify this as label filter with the list-key command. Even if you specify `"Wrap_Key\0"` with the --label option, the code in `tok_key_list_init()` still needs to find out the length of the string using `strlen()` which treats the first zero-byte as string end, and thus gets the wrong length for the filter. 

I currently do not see a way to find the 'real' string length of such a string.....
There is no other way to get the length of a string passed as program argument other than using `strlen()` or similar.

So lets keep it as it is for now. I guess I will open a new Issue to document this and maybe address at a later time.

Reply:
Opened new issue for this to not forget: https://github.com/opencryptoki/opencryptoki/issues/578